content/src/Blip.Api.Template/Controllers/ValidateController.cs

```
@@ -0,0 +1,298 @@
+        private readonly ISmallTalksDetector _smallTalksDetector;
+        private MySettings _settings;
+
+        public ValidateController(IDataValidator dataValidator, ILogger logger, IDateTimeValidator dateValidator, ISmallTalksDetector smallTalksDetector, MySettings settings)
+        {
+            _dataValidator = dataValidator;
+            _logger = logger;
+            _dateValidator = dateValidator;
+            _smallTalksDetector = smallTalksDetector;
+            _settings = settings;
+        }
+
+        [HttpPost("CpfCnpj")]
+        public async Task<IActionResult> ValidateCpfCnpjAsync(TextRequest request)
+        {
+            try
+            {
+                var isValid = _dataValidator.IsValid(request.Text, DataType.Cpf | DataType.Cnpj, out var resultAnalysis);
+                var documentType = resultAnalysis.UsedDataType.Equals(DataType.Cpf) ? DocumentType.CPF : resultAnalysis.UsedDataType.Equals(DataType.Cnpj) ? DocumentType.CNPJ : DocumentType.None;
+
+                return StatusCode(StatusCodes.Status200OK, new ValidateDocumentResponse { IsValid = isValid, Result = resultAnalysis.CleanedData, Type = documentType });
+            }
+            catch (Exception e)
+            {
+
+            }
+
+            return StatusCode(StatusCodes.Status500InternalServerError);
+
+        }
+
+        [HttpPost("Cpf")]
+        public async Task<IActionResult> ValidateCpfAsync(TextRequest request)
+        {
+            try
+            {
+                var isValid = _dataValidator.IsValid(request.Text, DataType.Cpf, out var resultAnalysis);
+                var documentType = resultAnalysis.UsedDataType.Equals(DataType.Cpf) ? DocumentType.CPF : DocumentType.None;
+
+                return StatusCode(StatusCodes.Status200OK, new ValidateDocumentResponse { IsValid = isValid, Result = resultAnalysis.CleanedData, Type = documentType });
+            }
+            catch (Exception e)
```

Again, **absolutelly NO empty catches**